src/shared/helpers.ts

```diff
@@ -193,6 +194,18 @@ export function bigIntMax(a: string, b: string) {
   return BigInt(a) > BigInt(b) ? a : b
 }
 
+export function tFactory(browser: Pick<Browser, 'i18n'>) {
+  type TranslationKeys = keyof typeof import('../_locales/en/messages.json')
+  /**
+   * Helper over calling cumbersome `this.browser.i18n.getMessage(key)` with
+   * added benefit that it type-checks if key exists in message.json
+   */
+  return <T extends TranslationKeys>(
+    key: T,
+    substitutions?: string | string[]
+  ) => browser.i18n.getMessage(key, substitutions)
+}
+
```

Comment:
Let's keep the types outside the function and export a `Translation` type.

```suggestion
type TranslationKeys = keyof typeof import('../_locales/en/messages.json')

export type Translation = ReturnType<typeof tFactory>

export function tFactory(browser: Pick<Browser, 'i18n'>) {
  /**
   * Helper over calling cumbersome `this.browser.i18n.getMessage(key)` with
   * added benefit that it type-checks if key exists in message.json
   */
  return <T extends TranslationKeys>(
    key: T,
    substitutions?: string | string[]
  ) => browser.i18n.getMessage(key, substitutions)
}

```